src/Components/RequestGeneratorBuilder.js

```diff
@@ -163,24 +170,34 @@ export default class RequestGeneratorBuilder extends preact.Component {
 
     render() {
         const replacers = replacer_factory(this);
-        const children_mapper = (c) =>
-            c.nodeName && Object.keys(replacers).includes(c.nodeName.name)
-                ? replacers[c.nodeName.name](c)
-                : Array.isArray(c.children)
-                ? { ...c, children: c.children.map(children_mapper) }
-                : c;
+        const children_mapper = (c) => {
+            if (!c || !c.type) return c;
+            if (Object.keys(replacers).includes(c.type.name)) {
+                return replacers[c.type.name](c);
+            }
+            if (c.props.children) {
+                c = cloneElement(c);
+                c.props.children =
+                    Array.isArray(c.props.children) ?
+                        c.props.children.map(children_mapper) :
+                        children_mapper(c.props.children);
+                return c;
+            }
+            return c;
+        }
         const children = this.props.children.map(children_mapper);
```

Comment:
I think the more idiomatic way to do this would be something like a render prop

https://reactjs.org/docs/render-props.html

(You pass in a function to generate the children and pass the components to that instead of receiving children and replacing placeholder)

Reply:
I've implemented this now